R/TealReportCard.R

```diff
@@ -43,21 +43,36 @@ TealReportCard <- R6::R6Class( # nolint: object_name_linter.
     #' card$get_content()[[1]]$get_content()
     #'
     append_fs = function(fs) {
-      checkmate::assert_list(fs)
-      attr_fs <- attr(fs, "formatted")
-      checkmate::assert_character(attr_fs, null.ok = TRUE)
-
+      # todo: whole class should be moved to teal
+      checkmate::assert_class(fs, "teal_slices")
       if (length(fs) != 0) {
+        states_list <- lapply(fs, function(x) {
+          x_list <- shiny::isolate(shiny::reactiveValuesToList(x))
+          if (
+            inherits(x_list$choices, c("integer", "numeric", "Date", "POSIXct", "POSIXlt")) &&
+              length(x_list$choices) == 2 &&
```

Comment:
When `choices` is of length 2, range-like variables are converted to factor and passed to `ChoicesFilterState`, so selected values should be enumerated.

Reply:
Yes, problem is that we have no information available which can help us deduct this scenario. We don't know the class of `FilterState`, just `teal_slice`. We don't know original `x` values. Also `"getOption("teal.threshold_slider_vs_checkboxgroup")`, doesn't say much.